File: common/src/main/java/de/werkelmann/rekote/model/HostInfo.java
package de.werkelmann.rekote.model;

import org.json.JSONException;
import org.json.JSONObject;

import java.net.InetAddress;
import java.net.UnknownHostException;

import de.werkelmann.rekote.util.RekoteException;

public class HostInfo {

    private final String hostName;
    private final String ipAddress;

    public HostInfo() throws RekoteException {
        try {
            this.hostName = InetAddress.getLocalHost().getHostName();
            this.ipAddress = InetAddress.getLocalHost().getHostAddress();
        } catch (UnknownHostException e) {
            throw new RekoteException("HostInfo could not get initialized");
        }
    }

    public HostInfo(JSONObject message) throws RekoteException {
        try {
            this.hostName = message.getString("hostName");
            this.ipAddress = message.getString("ipAddress");
        } catch (JSONException e) {
            throw new RekoteException("HostInfo could not get initialized");
        }
    }

    public String getHostName() {
        return hostName;
    }

    public String getIpAddress() {
        return ipAddress;
    }

    @Override
    public String toString() {
        return hostName + " " + ipAddress;
    }
}

File: common/src/main/kotlin/de/werkelmann/rekote/model/HostInfo.kt
package de.werkelmann.rekote.model

import de.werkelmann.rekote.util.RekoteException
import org.json.JSONException
import org.json.JSONObject

import java.net.InetAddress
import java.net.UnknownHostException

class HostInfo {

    val hostName: String
    val ipAddress: String

    @Throws(RekoteException::class)
    constructor() {
        try {
            this.hostName = InetAddress.getLocalHost().hostName
            this.ipAddress = InetAddress.getLocalHost().hostAddress
        } catch (e: UnknownHostException) {
            throw RekoteException("HostInfo could not get initialized")
        }

    }

    @Throws(RekoteException::class)
    constructor(message: JSONObject) {
        try {
            this.hostName = message.getString("hostName")
            this.ipAddress = message.getString("ipAddress")
        } catch (e: JSONException) {
            throw RekoteException("HostInfo could not get initialized")
        }

    }

    override fun toString(): String {
        return hostName + " " + ipAddress
    }
}
